src/diffusers/loaders/unet.py

```diff
@@ -374,6 +374,12 @@ def load_attn_procs(self, pretrained_model_name_or_path_or_dict: Union[str, Dict
             elif is_sequential_cpu_offload:
                 _pipeline.enable_sequential_cpu_offload()
             # Unsafe code />
+        elif is_custom_diffusion:
+                #Load custom diffusion cross attention weight with PEFT installed in environment
+                self.set_attn_processor(attn_processors)
+            
+                self.to(dtype=self.dtype, device=self.device)
```

Comment:
We're already doing this [here](https://github.com/huggingface/diffusers/blob/a1cb106459da4c595d22c04e026d7169d8dcfd2b/src/diffusers/loaders/unet.py#L363) no? 

Perhaps, guarding this code block with `if not is_custom_diffusion` would be more helpful? 

https://github.com/huggingface/diffusers/blob/a1cb106459da4c595d22c04e026d7169d8dcfd2b/src/diffusers/loaders/unet.py#L365

Reply:
Oh thanks so much. In that case, could we maybe simply move 

https://github.com/huggingface/diffusers/blob/a1cb106459da4c595d22c04e026d7169d8dcfd2b/src/diffusers/loaders/unet.py#L362

out of 

https://github.com/huggingface/diffusers/blob/a1cb106459da4c595d22c04e026d7169d8dcfd2b/src/diffusers/loaders/unet.py#L349

?

Reply:
I have tried this solution, but it failed. The reason is
https://github.com/huggingface/diffusers/blob/a1cb106459da4c595d22c04e026d7169d8dcfd2b/src/diffusers/loaders/unet.py#L369
will not be executed.

It will cause the bug "not same device" when you use GPU to train. (As the self is not been moved to GPU)
And I think 
https://github.com/huggingface/diffusers/blob/a1cb106459da4c595d22c04e026d7169d8dcfd2b/src/diffusers/loaders/unet.py#L369

can not be simply moved out of

https://github.com/huggingface/diffusers/blob/a1cb106459da4c595d22c04e026d7169d8dcfd2b/src/diffusers/loaders/unet.py#L349

since when PEFT is uninstalled,  it should between

https://github.com/huggingface/diffusers/blob/a1cb106459da4c595d22c04e026d7169d8dcfd2b/src/diffusers/loaders/unet.py#L350-L359

and 

https://github.com/huggingface/diffusers/blob/a1cb106459da4c595d22c04e026d7169d8dcfd2b/src/diffusers/loaders/unet.py#L371-L375

Hope to help:)

Reply:
I see. I understand now. 

I think the existing 

https://github.com/huggingface/diffusers/blob/a1cb106459da4c595d22c04e026d7169d8dcfd2b/src/diffusers/loaders/unet.py#L362

needs to be deleted no?

Reply:
Then it feels redundant. Because why would we want to add add the same piece of code twice? 😱

Reply:
Let's make this question clear.

Our target is to let the weight of cross-attention weights in custom diffusion load successfully whether PEFT is installed since
- PEFT is commonly installed for Lora
- but people who want to use custom_diffusion only don't need PEFT

In my opinion, we need to keep both pieces of code for
- PEFT offload UNet before 

https://github.com/huggingface/diffusers/blob/a1cb106459da4c595d22c04e026d7169d8dcfd2b/src/diffusers/loaders/unet.py#L349

so we don't need to offload UNet when PEFT is installed.

- If we delete it. This code may result in unsafety when PEFT is not installed since UNet is not offloaded

https://github.com/huggingface/diffusers/blob/a1cb106459da4c595d22c04e026d7169d8dcfd2b/src/diffusers/loaders/unet.py#L362-L363

Or maybe we can change this piece of code to:

```
        if not USE_PEFT_BACKEND:
            if _pipeline is not None:
                for _, component in _pipeline.components.items():
                    if isinstance(component, nn.Module) and hasattr(component, "_hf_hook"):
                        is_model_cpu_offload = isinstance(getattr(component, "_hf_hook"), CpuOffload)
                        is_sequential_cpu_offload = isinstance(getattr(component, "_hf_hook"), AlignDevicesHook)

                        logger.info(
                            "Accelerate hooks detected. Since you have called `load_lora_weights()`, the previous hooks will be first removed. Then the LoRA parameters will be loaded and the hooks will be applied again."
                        )
                        remove_hook_from_module(component, recurse=is_sequential_cpu_offload)

        # only custom diffusion needs to set attn processors
        if is_custom_diffusion:
            #Load custom diffusion cross attention weight with PEFT installed in environment
            self.set_attn_processor(attn_processors)

        # set lora layers
        for target_module, lora_layer in lora_layers_list:
            target_module.set_lora_layer(lora_layer)

        self.to(dtype=self.dtype, device=self.device)
        
        if not USE_PEFT_BACKEND:
            # Offload back.
            if is_model_cpu_offload:
                _pipeline.enable_model_cpu_offload()
            elif is_sequential_cpu_offload:
                _pipeline.enable_sequential_cpu_offload()
            # Unsafe code />
```

Reply:
Can't we leverage these two variables more systematically to process the logic you mentioned? `is_custom_diffusion` and `USE_PEFT_BACKEND`. I am uncomfortable keeping the same exact line of code like this as it makes the code confusing for us, maintainers.

Reply:
Your proposal looks good to me.

Reply:
OK, I will update my PR. It's nice to discuss with you, your idea about code inspires me a lot.